### DISCUSSION TITLE
Git lfs

### DISCUSSION THREAD

**Discussion by harmandeep2993** [community/community#140438] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

How to clear the lfs quota in order to upload new files.

**Comment by BDadmehr0** [comment]
To clear your LFS (Large File Storage) quota and upload new files, you'll generally want to follow these steps:

1. **Identify Unused Files**: Check which files in your LFS storage are no longer needed. You can do this by running:
   ```bash
   git lfs ls-files
   ```

2. **Remove Unused LFS Files**: To remove files that you no longer need, use:
   ```bash
   git lfs rm --cached <file>
   ```
   This command will untrack the file from LFS but keep it in your working directory.

3. **Delete Old LFS Objects**: If you want to permanently delete the objects from your LFS storage, use:
   ```bash
   git lfs prune
   ```
   This command will remove any LFS objects that are not referenced by any commits.

4. **Push Changes**: After removing the files and pruning, don’t forget to commit and push your changes:
   ```bash
   git commit -m "Remove unused LFS files"
   git push
   ```

5. **Check Quota**: If you’re using a service like GitHub, GitLab, or Bitbucket, you can check your current LFS usage in your repository settings.

6. **Upgrade Plan (if necessary)**: If you find that you consistently hit your quota, consider upgrading your LFS storage plan with your hosting provider.

Following these steps should help you clear out some space in your LFS quota and allow you to upload new files.

**Reply by harmandeep2993** [reply]
Hey,

Since, I have deleted the repositary already from the account. When tried to commit the changes it unable to process it.

I have files with *.dll and *:lib

**Comment by harmandeep2993** [comment]
Hey,

Since, I have deleted the repositary already from the account. When tried
to commit the changes it unable to process it.

I have files with *.dll and *:lib

**Reply by BDadmehr0** [reply]
If you've already deleted the repository from your account and are unable to commit changes because of the LFS quota, here’s what you can do:

### 1. **Clear Local LFS Cache**
Since the repository is deleted from the remote, you can clear your local LFS cache:

```bash
git lfs prune
```

This will remove any local LFS objects that are not referenced by any branches.

### 2. **Remove LFS Tracking**
If you want to remove LFS tracking for specific files locally, you can do it like this:

1. **Untrack the Files**:
   ```bash
   git lfs untrack "*.dll"
   git lfs untrack "*.lib"
   ```

2. **Remove the Entries from `.gitattributes`**:
   Open the `.gitattributes` file in your repository and remove any lines that include `*.dll` and `*.lib`.

### 3. **Commit the Changes**
Now, commit the changes to untrack these files:

```bash
git add .gitattributes
git commit -m "Remove LFS tracking for DLL and LIB files"
```

### 4. **Push to a New Repository (if needed)**
If you want to create a new repository and push the changes, you can do so:

1. Create a new repository on your Git hosting platform (GitHub, GitLab, etc.).
2. Set the new remote URL:
   ```bash
   git remote add origin <new-repo-url>
   ```
3. Push your changes:
   ```bash
   git push -u origin master
   ```

### 5. **Alternative File Management**
If you're not using LFS for these large files, consider adding them directly to your Git repository if the size allows.

If you encounter any specific errors, feel free to share them for further assistance.

**Comment by harmandeep2993** [comment]
git add .gitattributes
git commit -m "Remove LFS tracking for DLL and LIB files"

U       requirements.txt
error: Committing is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.

in conflict showing 10K + changes

**Comment by BDadmehr0** [comment]
It sounds like you have merge conflicts in your repository that need to be resolved before you can commit any changes. Here's how to handle this situation:

### Steps to Resolve Merge Conflicts:

1. **Identify Conflicted Files**:
   - Use the following command to see which files are in conflict:
     ```bash
     git status
     ```

2. **Open and Resolve Conflicts**:
   - Open each conflicted file in a text editor. Look for conflict markers (`<<<<<<<`, `=======`, `>>>>>>>`) which indicate the conflicting changes. You'll need to choose which changes to keep or merge them manually.
   - Remove the conflict markers once you've resolved the conflicts.

3. **Add Resolved Files**:
   - After resolving conflicts in a file, mark it as resolved by adding it:
     ```bash
     git add <file-name>
     ```
   - You can also add all resolved files:
     ```bash
     git add .
     ```

4. **Commit the Resolutions**:
   - Once all conflicts are resolved and added, commit the changes:
     ```bash
     git commit -m "Resolved merge conflicts"
     ```

5. **Continue with Your Previous Steps**:
   - After resolving the conflicts and committing, you can then proceed to remove LFS tracking and push your changes.

### Additional Tips:

- If you want to abort the merge process entirely, you can run:
  ```bash
  git merge --abort
  ```
  This will return your repository to the state before the merge attempt.

- If you are unsure about the changes to keep, consider using a diff tool or IDE with merge conflict resolution capabilities to help visualize the differences.

Let me know if you encounter any further issues or need clarification on any steps!
